CMakeLists.txt

```diff
@@ -373,6 +373,17 @@ if(MSVC)
     endif()
 endif()
 
+#
+# Additional flags for features checking
+#
+set(ADDITIONAL_CHECK_FLAGS )
+if(APPLE)
+    check_c_compiler_flag(-Werror=unguarded-availability-new HAVE_W_ERROR_UNGUARDED_AVAILABILITY_NEW)
+    if(HAVE_W_ERROR_UNGUARDED_AVAILABILITY_NEW)
+        set(ADDITIONAL_CHECK_FLAGS "-Werror=unguarded-availability-new")
+    endif()
+endif()
+
```

Comment:
For 10.10 and later, the flag should be `-Werror=unguarded-availability`, as the `-Werror=unguarded-availability-new` flag only supports 10.13 and later.

Reply:
Fixed